src/client.ts

```diff
@@ -5188,6 +5293,127 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
+            throw Error("Server does not support the Futures API");
+        }
+
+        const { threadId, eventType, content, txnId } = this.processEventArgs(
+            roomId,
+            threadIdOrEventType,
+            eventTypeOrContent,
+            contentOrTxnId,
+            txnIdOrVoid,
+        );
+
+        return this.sendCompleteEvent(roomId, threadId, { type: eventType, content }, futureOpts, txnId);
+    }
+
+    /**
+     * Send a future for a state event.
+     *
+     * Note: This endpoint is unstable, and can throw an `Error`.
+     *   Check progress on [MSC4140](https://github.com/matrix-org/matrix-spec-proposals/pull/4140) for more details.
+     */
+    // eslint-disable-next-line
+    public async _unstable_sendStateFuture<K extends keyof StateEvents, F extends ISendFutureRequestOpts>(
+        roomId: string,
+        futureOpts: F,
+        eventType: K,
+        content: StateEvents[K],
+        stateKey = "",
+        opts: IRequestOpts = {},
+    ): Promise<ISendFutureResponse<F>> {
+        if (!(await this.doesServerSupportUnstableFeature(UNSTABLE_MSC4140_FUTURES))) {
+            throw Error("Server does not support the Futures API");
+        }
+
+        const pathParams = {
+            $roomId: roomId,
+            $eventType: eventType,
+            $stateKey: stateKey,
+        };
+        let path = utils.encodeUri("/rooms/$roomId/state_future/$eventType", pathParams);
+        if (stateKey !== undefined) {
+            path = utils.encodeUri(path + "/$stateKey", pathParams);
+        }
+        return this.http.authedRequest(Method.Put, path, { ...futureOpts }, content as Body, {
+            prefix: `${ClientPrefix.Unstable}/${UNSTABLE_MSC4140_FUTURES}`,
+            ...opts,
+        });
+    }
+
+    /**
+     * Get all pending futures for the calling user.
+     *
+     * Note: This endpoint is unstable, and can throw an `Error`.
+     *   Check progress on [MSC4140](https://github.com/matrix-org/matrix-spec-proposals/pull/4140) for more details.
+     */
+    // eslint-disable-next-line
+    public async _unstable_getFutures(): Promise<IFutureInfo[]> {
+        if (!(await this.doesServerSupportUnstableFeature(UNSTABLE_MSC4140_FUTURES))) {
+            throw Error("Server does not support the Futures API");
+        }
+
+        return await this.http.authedRequest(Method.Get, "/future", undefined, undefined, {
+            prefix: `${ClientPrefix.Unstable}/${UNSTABLE_MSC4140_FUTURES}`,
+        });
+    }
+
+    /**
+     * Use a future token, taking the appropriate action given what the token is for.
+     *
+     * Note: This endpoint is unstable, and can throw an `Error`.
+     *   Check progress on [MSC4140](https://github.com/matrix-org/matrix-spec-proposals/pull/4140) for more details.
+     */
+    // eslint-disable-next-line
+    public async _unstable_useFutureToken(futureToken: string): Promise<{}> {
```

Comment:
`use` somehow implies the consequence of the token being invalid after use.
Which is not the case. What about `send`/`pingWith` or `updateFuture`/`updateFutureWithToken`.

Reply:
`updateFutureWithToken` is the most descriptive, so I'll go with that.